include/ur_api.h

```diff
@@ -7848,6 +7848,128 @@ typedef struct ur_callbacks_t {
     ur_device_callbacks_t Device;
 } ur_callbacks_t;
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Defines unique function identifiers
+typedef enum ur_function_t {
+    UR_FUNCTION_CONTEXT_CREATE = 1,
+    UR_FUNCTION_CONTEXT_RETAIN = 2,
+    UR_FUNCTION_CONTEXT_RELEASE = 3,
+    UR_FUNCTION_CONTEXT_GET_INFO = 4,
+    UR_FUNCTION_CONTEXT_GET_NATIVE_HANDLE = 5,
+    UR_FUNCTION_CONTEXT_CREATE_WITH_NATIVE_HANDLE = 6,
+    UR_FUNCTION_CONTEXT_SET_EXTENDED_DELETER = 7,
+    UR_FUNCTION_DEVICE_GET = 8,
+    UR_FUNCTION_DEVICE_GET_INFO = 9,
+    UR_FUNCTION_DEVICE_RETAIN = 10,
+    UR_FUNCTION_DEVICE_RELEASE = 11,
+    UR_FUNCTION_DEVICE_PARTITION = 12,
+    UR_FUNCTION_DEVICE_SELECT_BINARY = 13,
+    UR_FUNCTION_DEVICE_GET_NATIVE_HANDLE = 14,
+    UR_FUNCTION_DEVICE_CREATE_WITH_NATIVE_HANDLE = 15,
+    UR_FUNCTION_DEVICE_GET_GLOBAL_TIMESTAMPS = 16,
+    UR_FUNCTION_ENQUEUE_KERNEL_LAUNCH = 17,
+    UR_FUNCTION_ENQUEUE_EVENTS_WAIT = 18,
+    UR_FUNCTION_ENQUEUE_EVENTS_WAIT_WITH_BARRIER = 19,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_READ = 20,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_WRITE = 21,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_READ_RECT = 22,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_WRITE_RECT = 23,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_COPY = 24,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_COPY_RECT = 25,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_FILL = 26,
+    UR_FUNCTION_ENQUEUE_MEM_IMAGE_READ = 27,
+    UR_FUNCTION_ENQUEUE_MEM_IMAGE_WRITE = 28,
+    UR_FUNCTION_ENQUEUE_MEM_IMAGE_COPY = 29,
+    UR_FUNCTION_ENQUEUE_MEM_BUFFER_MAP = 31,
+    UR_FUNCTION_ENQUEUE_MEM_UNMAP = 33,
```

Comment:
Missing 30 and 32, isn't this enum autogenerated?

Reply:
I had to rebase this many times now, which is very fiddly... In general, ids don't have to be monotonically increasing.

Reply:
Turns out it works fine, entry points with ids 30 and 32 are disabled, see #50.